supereeg/helpers.py

```diff
@@ -506,27 +497,48 @@ def _timeseries_recon(bo, K, chunk_size=1000, preprocess='zscore'):
             data = bo.get_data().as_matrix()
         else:
             data = bo.get_zscore_data()
-
-    s = K.shape[0] - data.shape[1]
-    Kba = K[:s, s:]
-    Kaa = K[s:, s:]
+    else:
+        raise('Unsupported preprocessing option: ' + preprocess)
+
+    brain_locs_in_model = _count_overlapping(mo.get_locs(), bo.get_locs())
+    model_locs_in_brain = _count_overlapping(bo.get_locs(), mo.get_locs())
+
+    if np.all(model_locs_in_brain):
+        #if the model contains all of the locations (or fewer) than what are in the brain object, no reconstructions
+        #are needed
+        return data[:, brain_locs_in_model]
+
+    #otherwise, we'll need to do some work
+    Z = mo.get_model(z_transform=True)
+    if ~np.any(brain_locs_in_model):
+        #if none of the brain locations are in the model, we need to blur out the model to match up with the
+        # locations in the brain object
+        combined_locs = np.vstack((bo.get_locs(), mo.get_locs()))
+        model_locs_in_brain = [False]*bo.get_locs().shape[0]
+        model_locs_in_brain.extend([True]*mo.get_locs().shape[0])
+
+        rbf_weights = _log_rbf(combined_locs, mo.get_locs())
+        Z = _blur_corrmat(Z, rbf_weights)
+
```

Comment:
Observation from @lucywowen: what if the brain locations *partially* overlap with the model locations?

Reply:
```
if ~np.all(brain_locs_in_model):
    combined_locs = np.vstack((bo.get_locs(), mo.get_locs()))
    if np.any(brain_locs_in_model):
        Combined_locs = np.unique_rows(combined_locs)
        < update model_locs_in_brain>
    Else:
        Model_locs_in_brain = [False]*bo.get_locs().shape[0]
        Model_locs_in_brain.extend([True]*mo.get_locs().shape[0])
    Rbf_weights = ...
    Z = ...